probe-rs/src/flashing/loader.rs

```diff
@@ -136,6 +137,29 @@ impl FlashLoader {
         let mut buf = Vec::new();
         file.read_to_end(&mut buf)?;
 
+        // Figure out flash size from the memory map. We need a different bootloader for each size.
+        let flash_size = match target.debug_sequence.clone() {
+            DebugSequence::Riscv(sequence) => sequence
+                .detect_flash_size(session.get_riscv_interface().unwrap())
```

Comment:
Regarding error handling here, I think that we should ignore (but maybe print a warn) any errors when trying to detect the flash size.